### PR TITLE
LMB-520 | Create person from mandataris

### PR DESCRIPTION
**Pull request by JonasVanHoof**
## Description

When the person is not found in database it needs to be created. Check the first and lastname on the mandataris subject itself. Persons are not being consumed by the BnB harvaster.

## How to test

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration.

## Links to other PR's

- https://github.com/lblod/frontend-lokaal-mandatenbeheer/pull/180

### CODE REVIEW

File: services/processing-queue.ts [hunk 1]
```diff
@@ -5,73 +5,69 @@ export class ProcessingQueue {
...
+        console.log('|> TASK done \n|>\n');
         this.run();
       }
-    } else {
-      setTimeout(() => {
-        console.log(
-          `|> Trigger run, ${this.queue.length} remaining queue items and ${this.manualQueue.length} manual queued items`,
-        );
-        this.run();
-      }, this.queuePollInterval);
```
Comment (JonasVanHoof) [inline services/processing-queue.ts:41]: This is removed by a trigger by action. This also makes that it is not running the setTimeout when nothing is triggered in an hour, which is good :D 

Result log

```bash
mandataris-1  | 2024-07-03T14:45:01.505693488Z |> Trigger endpoint mandatees-decisions
mandataris-1  | 2024-07-03T14:45:01.505836945Z |> CURRENT QUEUE ITEMS 0 and 0 in manual queue
mandataris-1  | 2024-07-03T14:45:01.509154185Z |> Moving 240 items from manual queue to the acutal executing queue.
mandataris-1  | 2024-07-03T14:45:01.509196090Z |> [2024-07-03T14:45:01.505Z] Added 240 to queue.
mandataris-1  | 2024-07-03T14:45:01.509202969Z |> Currently 240 items in queue.
mandataris-1  | 2024-07-03T14:45:01.509208565Z |> Queue was not empty triggering run()
mandataris-1  | 2024-07-03T14:45:01.509228923Z |> TASK start for http://data.lblod.info/id/mandatarissen/2743c2dc-818d-42aa-b4c0-1e972a53e18f

```

File: controllers/mandatees-decisions.ts [hunk 2]
```diff
@@ -105,8 +117,23 @@ export async function handleTriplesForMandatarisSubject(
...
-    console.log('|> Persoon does not exist: TODO in LMB-520');
+    const persoon = await findNameOfPersoonFromStaging(mandatarisSubject);
+    console.log('|> Looking for persoon names', persoon);
+    if (!persoon || (!persoon.firstname && !persoon.lastname)) {
+      mandatarisQueue.addToManualQueue(mandatarisSubject);
+      return;
+    }
+
+    console.log('|> ... creating persoon');
+    const RRN = '';
```
[the remark this answers is not in the document]
Comment (JonasVanHoof) [reply controllers/mandatees-decisions.ts:128]: Create new method

File: data-access/mandatees-decisions.ts [hunk 3]
```diff
@@ -305,3 +298,29 @@ export async function insertTriplesInGraph(
...
+  const queryMandatarisPerson = `
+    PREFIX foaf: <http://xmlns.com/foaf/0.1/>
+    PREFIX persoon: <https://data.vlaanderen.be/ns/persoon#>
+
+    SELECT ?firstname ?lastname
+    WHERE {
+      GRAPH ${sparqlEscapeTermValue(TERM_STAGING_GRAPH)} {
+        ${mandatarisUri} a <http://data.vlaanderen.be/ns/mandaat#Mandataris> .
+                      
+        OPTIONAL {
```
[the remark this answers is not in the document]
Comment (JonasVanHoof) [reply data-access/mandatees-decisions.ts:315]: Todo: remove optionals

File: controllers/mandatees-decisions.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (bfidlers) [reply controllers/mandatees-decisions.ts:128]: You can just reuse the uri of the persoon, as this uri is not yet used in our database, that way it is already linked.

**Review by bfidlers — APPROVED** [review]

**Pull request merged**